workflow/controller/workflowpod.go

```diff
@@ -139,6 +139,14 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 		pod.Spec.Containers = append(pod.Spec.Containers, *waitCtr)
 	}
 
+	if tmpl.GetType() == wfv1.TemplateTypeContainer || tmpl.GetType() == wfv1.TemplateTypeScript {
```

Comment:
How about putting this logic in `addSchedulingConstraints` as other scheduling constraints are generated there?

Reply:
Will do, thanks